hack/lib/golang.sh

```diff
@@ -680,7 +680,7 @@ kube::golang::build_binaries() {
     # Use eval to preserve embedded quoted strings.
     local goflags goldflags goasmflags gogcflags
     eval "goflags=(${GOFLAGS:-})"
-    goldflags="${GOLDFLAGS:-} -s -w $(kube::version::ldflags)"
+    goldflags="${GOLDFLAGS:-"-s -w"} $(kube::version::ldflags)"
```

Comment:
I like having an overridable default, but this doesn't seem to work as currently written - `${var:-default}` results in the default value being used either when `var` is undefined or when it's null, so setting `GOLDFLAGS=` won't help here.

If you change it to `${GOLDFLAGS-"-s -w"}`, that won't work either, since it is always defined inside the build image (https://github.com/kubernetes/kubernetes/blob/06e737367d0b53b59e64e464f4a3a2393a40b7d7/build/common.sh#L568).

Reply:
hmmm. good point. basically the only use case that isn't covered is "want to set GOLDFLAGS to a null value"